pkgs/applications/misc/phoc/default.nix

```diff
@@ -0,0 +1,68 @@
+{ stdenv
+, fetchFromGitLab
+, meson
+, ninja
+, pkgconfig
+, wrapGAppsHook
+, libinput
+, gnome3
+, glib
+, gtk3
+, wayland
+, dbus
+, cmake
+, libdrm
+, libxkbcommon
+, wlroots
+}:
+
+stdenv.mkDerivation rec {
+  pname = "phoc";
+  version = "0.1.7";
+
+  src = fetchFromGitLab {
+    domain = "source.puri.sm";
+    owner = "Librem5";
+    repo = pname;
+    rev = "3d68b091d0075bd549f6fbb84d6c152f01a2354e";
+    sha256 = "1007kxap87h1bi79sz45lfb478d93pq0676cr5rnhbanljm1n28n";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    meson
+    ninja
+    pkgconfig
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    libdrm.dev
+    libxkbcommon
+    libinput
+    glib
+    gtk3
+    gnome3.gnome-desktop
+    gnome3.mutter
+    wayland
+    wlroots
+  ];
+
+  NIX_CFLAGS_COMPILE = "-I${libdrm.dev}/include/libdrm/";
+
+  configurePhase = "meson build --prefix=$out";
```

Comment:
these phases should be handled properly thanks to `pkgs/development/tools/build-managers/meson/setup-hook.sh` and `pkgs/development/tools/build-managers/ninja/setup-hook.sh`.

Reply:
The same happens for phoc:

```
building '/nix/store/fal6pfjjhz58y2xav97i59z3fzxqz293-phoc-0.1.7.drv'...
unpacking sources
unpacking source archive /nix/store/57z50rdws7643if0l17hna8hvznfyy4l-source
source root is source
patching sources
configuring
fixing cmake files...
cmake flags: -GNinja -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/5ivc75pr4gh6vm1kx55qzhgjxqkvkfkd-phoc-0.1.7/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/5ivc75pr4gh6vm1kx55qzhgjxqkvkfkd-phoc-0.1.7/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/5ivc75pr4gh6vm1kx55qzhgjxqkvkfkd-phoc-0.1.7/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/5ivc75pr4gh6vm1kx55qzhgjxqkvkfkd-phoc-0.1.7/share/doc/ -DCMAKE_INSTALL_INFODIR=/nix/store/5ivc75pr4gh6vm1kx55qzhgjxqkvkfkd-phoc-0.1.7/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/5ivc75pr4gh6vm1kx55qzhgjxqkvkfkd-phoc-0.1.7/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/5ivc75pr4gh6vm1kx55qzhgjxqkvkfkd-phoc-0.1.7/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/5ivc75pr4gh6vm1kx55qzhgjxqkvkfkd-phoc-0.1.7/include -DCMAKE_INSTALL_SBINDIR=/nix/store/5ivc75pr4gh6vm1kx55qzhgjxqkvkfkd-phoc-0.1.7/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/5ivc75pr4gh6vm1kx55qzhgjxqkvkfkd-phoc-0.1.7/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/5ivc75pr4gh6vm1kx55qzhgjxqkvkfkd-phoc-0.1.7/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_OSX_ARCHITECTURES=x86_64 -DCMAKE_FIND_FRAMEWORK=last -DCMAKE_STRIP=/nix/store/a57856fs4m8ir6vlv14h3gq3sv9aq2lb-binutils-2.31.1/bin/strip -DCMAKE_RANLIB=/nix/store/a57856fs4m8ir6vlv14h3gq3sv9aq2lb-binutils-2.31.1/bin/ranlib -DCMAKE_AR=/nix/store/a57856fs4m8ir6vlv14h3gq3sv9aq2lb-binutils-2.31.1/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/5ivc75pr4gh6vm1kx55qzhgjxqkvkfkd-phoc-0.1.7
CMake Error: The source directory "/build/source" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
builder for '/nix/store/fal6pfjjhz58y2xav97i59z3fzxqz293-phoc-0.1.7.drv' failed with exit code 1
```

Reply:
removing `cmake` produces this error:
```
...

Library m found: YES
Git program not found.
Subproject  subprojects/wlroots is buildable: NO (disabling)

meson.build:50:8: ERROR: Problem encountered: Wlroots subproject not found but enabled
```

Reply:
With CMake and meson, we need to disable the setup hook with

```nix
dontUseCmakeConfigure = true;
```

Reply:
solved! :)